File: Demo/Sources/ViewControllers/DataView/List/ListDataViewController.m
//
//  Globus
//

#import "ListDataViewController.h"
#import "ListDataViewCell.h"
#import "ListDataProvider.h"

@interface ListDataViewController ()
@end

static NSString* ListCellIdentifier = @"List";

@implementation ListDataViewController

#pragma mark - Init / Free

- (void)setup {
    [super setup];
    
    self.edgesForExtendedLayout = UIRectEdgeNone;
    self.title = @"ListDataView";
    
    self.provider = [ListDataProvider dataProvider];
    self.spinnerView = [GLBArcSpinnerView new];
    self.spinnerView.color = UIColor.blueColor;
}

#pragma mark - GLBViewController

- (NSArray< UIBarButtonItem* >*)prepareNavigationLeftBarButtons {
    return @[
        [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"MenuButton"] style:UIBarButtonItemStylePlain target:self action:@selector(pressedMenu:)]
    ];
}

#pragma mark - GLBListDataViewController

- (GLBDataViewContainer< GLBListDataViewControllerContentContainerProtocol >*)prepareContentContainer {
    return [ListDataViewControllerContentContainer container];
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

@end

@implementation ListDataViewControllerContentContainer

+ (NSDictionary< NSString*, Class >*)mapCells {
    return @{
        ListCellIdentifier: ListDataViewCell.class
    };
}

- (GLBDataViewItem*)prepareItemWithModel:(id)model {
    return [GLBDataViewItem itemWithIdentifier:ListCellIdentifier order:0 data:model];
}

@end

File: Demo/Sources/ViewControllers/DataView/List/ListDataViewController.swift
//
//  Globus
//

import Globus

class ListDataViewController: GLBListDataViewController {
    
    // MARK: - Init / Free
    
    override func setup() {
        super.setup()
        
        self.edgesForExtendedLayout = UIRectEdge.init(rawValue: 0);
        self.title = "ListDataView"
        
        self.provider = ListDataProvider();
        self.spinnerView = GLBArcSpinnerView();
        self.spinnerView!.color = UIColor.blue;
    }
    
    // MARK: - GLBViewController
    
    override func prepareNavigationLeftBarButtons() -> [UIBarButtonItem] {
        return [
            UIBarButtonItem(image: UIImage(named: "MenuButton"), style: .plain, target: self, action: #selector(pressedMenu(_ :)))
        ]
    }
    
    // MARK: - GLBListDataViewController
    
    override func prepareContentContainer() -> GLBDataViewContainer? {
        return ListDataViewControllerContentContainer()
    }
    
    // MARK: - Actions
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
}

class ListDataViewControllerContentContainer: GLBListDataViewControllerContentContainer {
    
    // MARK: - GLBListDataViewControllerContentContainer
    
    override class func mapCells() -> [String: AnyClass] {
        return [
            Const.ListCellIdentifier: ListDataViewCell.self
        ]
    }
    
    override func prepareItem(withModel model: Any) -> GLBDataViewItem?{
        return GLBDataViewItem.init(identifier: Const.ListCellIdentifier, order: 0, data: model)
    }
    
    // MARK: - Const
    
    struct Const {
        static let ListCellIdentifier = "List"
    }
    
}
